src/modules/hip/kernel/resample.hpp

```diff
@@ -0,0 +1,171 @@
+    extern __shared__ float windowCoeffs_smem[];
+    float *windowCoeffsSh = windowCoeffs_smem;
+    for (int k = hipThreadIdx_x; k < window.lookupSize; k += block)
+        windowCoeffsSh[k] = window.lookup[k];
+    __syncthreads();
+    window.lookup = windowCoeffsSh;
+
+    int outBlock = id_x * block;
+    if (outBlock >= outEnd)
+        return;
+
+    int blockEnd = std::min(outBlock + block, outEnd);
+    double inBlockRaw = outBlock * scale;
+    int inBlockRounded = static_cast<int>(inBlockRaw);
+    float inPos = inBlockRaw - inBlockRounded;
+    float fscale = scale;
+
+    for (int outPos = outBlock; outPos < blockEnd; outPos++, inPos += fscale)
+    {
+        int loc0, loc1;
+        window.input_range(inPos, &loc0, &loc1);
+
+        if (loc0 + inBlockRounded < 0)
+            loc0 = -inBlockRounded;
+        if (loc1 + inBlockRounded > srcLength)
+            loc1 = srcLength - inBlockRounded;
+        int locInWindow = loc0;
+        float locBegin = locInWindow - inPos;
+        float accum = 0.0f;
+
+        for (; locInWindow < loc1; locInWindow++, locBegin++)
+        {
+            float w = window(locBegin);
+            accum += srcPtr[inBlockRounded + locInWindow] * w;
+        }
+        dstPtr[outPos] = accum;
+    }
+}
+
+// Generic n channel resample support
+__global__ void resample_nchannel_hip_tensor(float *srcPtr,
+                                             float *dstPtr,
+                                             int2 srcDims,
+                                             int outEnd,
+                                             double scale,
+                                             RpptResamplingWindow &window,
+                                             int block)
+{
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+
+    int outBlock = id_x * block;
+    if (outBlock >= outEnd)
+        return;
+
+    int blockEnd = std::min(outBlock + block, outEnd);
+    double inBlockRaw = outBlock * scale;
+    int inBlockRounded = static_cast<int>(inBlockRaw);
+    float inPos = inBlockRaw - inBlockRounded;
+    float fscale = scale;
+    float tempBuf[RPPT_MAX_CHANNELS] = {0.0f}; // Considering max channels as 3
+
+    for (int outPos = outBlock; outPos < blockEnd; outPos++, inPos += fscale)
+    {
+        int loc0, loc1;
+        window.input_range(inPos, &loc0, &loc1);
+
+        if (loc0 + inBlockRounded < 0)
+            loc0 = -inBlockRounded;
+        if (loc1 + inBlockRounded > srcDims.x)
+            loc1 = srcDims.x - inBlockRounded;
+        int locInWindow = loc0;
+        float locBegin = locInWindow - inPos;
+        int2 ofs_i2 = make_int2(loc0, loc1);
+        ofs_i2 *= (int2)srcDims.y;
+        int idx = inBlockRounded * srcDims.y;
+
+        for (int inOfs = ofs_i2.x; inOfs < ofs_i2.y; inOfs += srcDims.y, locBegin++)
+        {
+            float w = window(locBegin);
+            for (int c = 0; c < srcDims.y; c++)
+                tempBuf[c] += srcPtr[idx + inOfs + c] * w;
```

Comment:
Again @snehaa8 idx + inOfs is common, we should be taking it out of both loops and not adding looped compute instructions. In the current situation, say idx=5, ofs_i2.x=3 and srcDims.y=7,

idx + inofs:
5 + (3) = 8 -> srcDims.y times
5 + (3 + 7) = 14 -> srcDims.y times
5 + (9 + 7) = 21 -> srcDims.y times

Please just make idx part of the loop and just add inofs to it as part of the loop increment, then you dont need to do an idx + inOfs anywhere

locInWindow is used just once so please refrain from using additional variables unless used repeatedly.
```
float locBegin = loc0 - inPos;
```